pkgs/by-name/cu/cup-docker/package.nix

```diff
@@ -0,0 +1,98 @@
+{
+  rustPlatform,
+  fetchFromGitHub,
+  lib,
+  stdenvNoCC,
+  bun,
+  nodejs-slim_latest,
+  nix-update-script,
+  withServer ? true,
+}:
+let
+  pname = "cup-docker";
+  version = "3.4.0";
+  src = fetchFromGitHub {
+    owner = "sergi0g";
+    repo = "cup";
+    tag = "v${version}";
+    hash = "sha256-ciH3t2L2eJhk1+JXTqEJSuHve8OuVod7AuQ3iFWmKRE=";
+  };
+  web = stdenvNoCC.mkDerivation (finalAttrs: {
+    pname = "${pname}-web";
+    inherit version src;
+    impureEnvVars = lib.fetchers.proxyImpureEnvVars ++ [
+      "GIT_PROXY_COMMAND"
+      "SOCKS_SERVER"
+    ];
+    sourceRoot = "${finalAttrs.src.name}/web";
+    nativeBuildInputs = [
+      bun
+      nodejs-slim_latest
+    ];
+    configurePhase = ''
+      runHook preConfigure
+      bun install --no-progress --frozen-lockfile
+      substituteInPlace node_modules/.bin/{vite,tsc} \
+        --replace-fail "/usr/bin/env node" "${nodejs-slim_latest}/bin/node"
```

Comment:
This wouldn't be necessary if done with a proper builder.

Reply:
Neat, which one? I'm not familiar with bun and I'd love to learn, is there a existing builder that applies?